### ISSUE TITLE
Issue with d3 and text, composing d3 objects in general

### ISSUE THREAD

**Issue by ianks** [CSCI-4830-002-2014/challenge-week-8#24]
I made a post on StackOverflow and had some questions:

http://stackoverflow.com/questions/26472294/d3-attach-text-to-circle-such-that-it-has-same-priority-as-circle-object

**Comment by doubleshow** [comment]
Challenge 2 is designed to show you how to use "g" to group components together so that they are composable. Take a look at the skeleton code. Does this answer your question? Or did you mean to ask something different?

**Comment by ianks** [comment]
Yes thanks for that. I am also concerned with the idea of reusability in general with d3, and haven't stumbled on the best way to do that yet. I found an interesting article here covering some best practices for reusability:

http://bocoup.com/weblog/reusability-with-d3/
